File: 2023/advent_of_code/test/q17/q17_test.exs
defmodule AdventOfCode.Q17Test do
  use ExUnit.Case
  doctest AdventOfCode.Q17

  test "Q17 Part 1 Example works" do
    file_stream = File.stream!("./test/q17/q17_example.txt")
    result = AdventOfCode.Q17.part1(file_stream)
    assert result == 102
  end

  # Taking roughly 5 min on my M1 Mac
  @tag timeout: 600_000
  test "Q17 Part 1 works" do
    file_stream = File.stream!("./test/q17/q17.txt")
    result = AdventOfCode.Q17.part1(file_stream)
    assert result == 1256
  end

  test "Q17 Part 2 Example works" do
    file_stream = File.stream!("./test/q17/q17_example.txt")
    result = AdventOfCode.Q17.part2(file_stream)
    assert result == 94
  end

  test "Q17 Part 2 Example 2 works" do
    file_stream = File.stream!("./test/q17/q17_example.txt")
    result = AdventOfCode.Q17.part2(file_stream)
    assert result == 71
  end

  # Taking roughly 5 min on my M1 Mac
  @tag timeout: 600_000
  test "Q17 Part 2 works" do
    file_stream = File.stream!("./test/q17/q17.txt")
    result = AdventOfCode.Q17.part2(file_stream)
    assert result == 1382
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q17/Q17Test.kt
package q17

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q17Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q17.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(3161, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q17.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(1575931232076, part2(filePath!!))
    }
}